lib/rdoc/markup/to_html_crossref.rb

```diff
@@ -156,16 +156,26 @@ def link(name, text, code = true, rdoc_ref: false)
     ref = @cross_reference.resolve name, text if name
 
     case ref
-    when String then
+    when String
       if rdoc_ref && @options.warn_missing_rdoc_ref
         puts "#{@from_path}: `rdoc-ref:#{name}` can't be resolved for `#{text}`"
       end
       ref
     else
       path = ref ? ref.as_href(@from_path) : +""
 
-      if code and RDoc::CodeObject === ref and !(RDoc::TopLevel === ref)
-        text = "<code>#{CGI.escapeHTML text}</code>"
+      text = if code && RDoc::TopLevel === ref
+        # Allow explicit rdoc-ref links to files, but don't wrap in code tags
+        if rdoc_ref
+          text
+        else
+          # Don't auto-link file paths in backticks
+          return "<code>#{CGI.escapeHTML text}</code>"
```

Comment:
If the README.md content is
```markdown
README `README`
```
Two `"README"` will both trigger `link("README", "README", true, false)`.
One from `handle_regexp_CROSSREF` → `cross_reference` → `link`
Another from `convert_tt_crossref` → `cross_reference` → `link`
So if we need to detect backticks, we need to pass a new argument from `convert_tt_crossref`.

Reply:
For this pull request, we need to fix this.

```markdown
See README.md `README.md`
```
<img width="857" height="194" alt="image" src="https://github.com/user-attachments/assets/084eacdd-6d39-4998-99b3-d2d243d56ec1" />
Only filenames inside backtick should use `<code>` but currently filenames outside backtick also converted to `<code>`.